syntax = "proto3";

message swarm_error
{
    string message = 1;
    bytes data = 2;
}


message bzn_envelope
{
    string swarm_id = 1;
    string sender = 2;
    bytes signature = 3;
    uint64 timestamp = 4;
    repeated bzn_envelope piggybacked_requests = 5;

    oneof payload
    {
        bytes database_msg = 10;
        bytes pbft_internal_request = 11;
        bytes database_response = 12;
        bytes json = 13;
        bytes audit = 14;
        bytes pbft = 15;
        bytes pbft_membership = 16;
        bytes status_request = 17;
        bytes status_response = 18;
        bytes checkpoint_msg = 19;
        bytes swarm_error = 20;
    }
}
